File: Utilities/SqlScriptRunner/ScriptSources/ScriptSource.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using SqlScriptRunner.ScriptTypes;
namespace SqlScriptRunner.ScriptSources
{
	abstract class ScriptSource
	{
		internal const string DataScriptDir = "Data Scripts";
		internal const string UpdateScriptDir = "Update Scripts";
		internal const string CreateScriptDir = "Create Scripts";
		internal const string OnSyncScriptDir = "OnSync Scripts";

		internal const string ProcedureSubDir = "prc";
		internal const string FunctionSubDir = "udf";
		internal const string TriggerSubDir = "trg";
		internal const string ViewSubDir = "viw";



		List<ObjectScript> functionScripts = new List<ObjectScript>();
		List<ObjectScript> procedureScripts = new List<ObjectScript>();
		List<ObjectScript> viewScripts = new List<ObjectScript>();
		List<ObjectScript> triggerScripts = new List<ObjectScript>();
		SortedList<int, UpdateScript> updateScripts = new SortedList<int, UpdateScript>();
		List<DataScript> tableDataScripts = new List<DataScript>();
		List<OnSyncScript> onSyncScripts = new List<OnSyncScript>();





		internal List<ObjectScript> FunctionScripts
		{
			get
			{
				return this.functionScripts;
			}
		}

		internal List<ObjectScript> ProcedureScripts
		{
			get
			{
				return this.procedureScripts;
			}
		}

		internal List<DataScript> TableDataScripts
		{
			get
			{
				return this.tableDataScripts;
			}
		}

		internal List<ObjectScript> TriggerScripts
		{
			get
			{
				return this.triggerScripts;
			}
		}

		internal List<UpdateScript> UpdateScripts
		{
			get
			{
				return new List<UpdateScript>(this.updateScripts.Values);
			}
		}

		internal List<ObjectScript> ViewScripts
		{
			get
			{
				return this.viewScripts;
			}
		}
		internal List<OnSyncScript> OnSyncScripts
		{
			get
			{
				return this.onSyncScripts;
			}
		}

		internal void AddScript(string type, string path)
		{

			switch (type)
			{
				case ScriptType.TableData:
					this.tableDataScripts.Add(new DataScript(path)); break;
				case ScriptType.UpdateScript:
					UpdateScript updateScript = new UpdateScript(path);
					this.updateScripts.Add(updateScript.VersionNumber, updateScript); break;
				case ScriptType.Trigger:
					this.triggerScripts.Add(new ObjectScript(path, type)); break;
				case ScriptType.StoredProcedure:
					this.procedureScripts.Add(new ObjectScript(path, type)); break;
				case ScriptType.UserDefinedFunction:
					this.functionScripts.Add(new ObjectScript(path, type)); break;
				case ScriptType.View:
					this.viewScripts.Add(new ObjectScript(path, type)); break;
				case ScriptType.OnSync:
					this.onSyncScripts.Add(new OnSyncScript(path)); break;
				default: break;
			}


		}

	}
}

File: Utilities/DatabaseUpdater/ScriptSource.vb
Imports System.Collections.Generic
Public MustInherit Class ScriptSource
	Friend _functionScripts As New List(Of ObjectScript)
	Friend _procedureScripts As New List(Of ObjectScript)
	Friend _viewScripts As New List(Of ObjectScript)
	Friend _triggerScripts As New List(Of ObjectScript)
	Friend _updateScripts As New List(Of UpdateScript)
	Friend _tableDataScripts As New List(Of TableDataScript)

	Public MustOverride Sub FillTables()




	Public ReadOnly Property FunctionScripts() As System.Collections.Generic.List(Of ObjectScript)
		Get
			Return Me._functionScripts
		End Get
	End Property

	Public ReadOnly Property ProcedureScripts() As System.Collections.Generic.List(Of ObjectScript)
		Get
			Return Me._procedureScripts
		End Get
	End Property

	Public ReadOnly Property TableDataScripts() As System.Collections.Generic.List(Of TableDataScript)
		Get
			Return Me._tableDataScripts
		End Get
	End Property

	Public ReadOnly Property TriggerScripts() As System.Collections.Generic.List(Of ObjectScript)
		Get
			Return Me._triggerScripts
		End Get
	End Property

	Public ReadOnly Property UpdateScripts() As System.Collections.Generic.List(Of UpdateScript)
		Get
			Return Me._updateScripts
		End Get
	End Property

	Public ReadOnly Property ViewScripts() As System.Collections.Generic.List(Of ObjectScript)
		Get
			Return Me._viewScripts
		End Get
	End Property

	Sub AddScript(ByVal type As String, ByVal path As String)
		Select Case type
			Case ScriptType.TableData
				Me._tableDataScripts.Add(New TableDataScript(path))
			Case ScriptType.UpdateScript
				Me._updateScripts.Add(New UpdateScript(path))
			Case ScriptType.Trigger
				Me._triggerScripts.Add(New ObjectScript(path, type))
			Case ScriptType.StoredProcedure
				Me._procedureScripts.Add(New ObjectScript(path, type))
			Case ScriptType.UserDefinedFunction
				Me._functionScripts.Add(New ObjectScript(path, type))
			Case ScriptType.View
				Me._viewScripts.Add(New ObjectScript(path, type))

		End Select

	End Sub
End Class

